package.json

```diff
@@ -31,48 +31,63 @@
     "url": "git+https://github.com/MakerXStudio/graphql-apollo-server.git"
   },
   "scripts": {
-    "build": "run-s build:*",
-    "build:0-clean": "rimraf dist coverage",
-    "build:1-compile": "rollup -c --configPlugin typescript",
-    "build:2-copy-pkg-json": "npx --yes @makerx/ts-toolkit@latest copy-package-json --custom-sections type module types exports",
-    "build:3-copy-readme": "copyfiles README.md dist",
-    "test": "jest --passWithNoTests --coverage --coverageReporters=text",
+    "audit": "better-npm-audit audit",
+    "check-types": "tsc --noEmit",
     "lint": "eslint ./src/ --ext .ts",
     "lint:fix": "eslint ./src/ --ext .ts --fix",
-    "check-types": "tsc --noEmit",
-    "audit": "better-npm-audit audit"
+    "test": "vitest run",
+    "test:coverage": "vitest run --coverage",
+    "test:ci": "vitest run --coverage --reporter junit --outputFile test-results.xml",
+    "codegen": "graphql-codegen --config codegen-testing.ts",
+    "build": "run-s build:*",
+    "build:0-clean": "rimraf dist coverage",
+    "build:1-build": "rollup -c --configPlugin typescript",
+    "build:2-generate-types": "tsc -p tsconfig.json",
+    "build:3-copy-pkg-json": "tstk copy-package-json -c",
+    "build:4-copy-readme": "copyfiles README.md dist"
   },
   "dependencies": {
-    "@makerx/graphql-core": "^2.2.0",
+    "@makerx/graphql-core": "^2.2.2",
     "@makerx/node-common": "^1.4.0"
   },
   "peerDependencies": {
     "@apollo/server": "4 || 5",
+    "@graphql-typed-document-node/core": "*",
```

Comment:
This would generally be referenced by the any parent project running codegen, hence it being a peer dep